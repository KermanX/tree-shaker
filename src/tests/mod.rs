use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
