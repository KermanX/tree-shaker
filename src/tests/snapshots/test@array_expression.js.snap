---
source: src/tests/mod.rs
assertion_line: 32
expression: tree_shake(input)
input_file: src/tests/fixtures/array_expression.js
---
export function main(unknown) {
	effect(2, "ab");
	const arr2 = [
		4,
		...[6],
		,
		...unknown,
		unknown,
		...[...[1, 4]]
	];
	effect(arr2);
	const __unused_2782 = effect();
	const __unused_F2EA = effect();
	const __unused_ACA8 = [effect(), effect()];
}
export function test2() {
	function f() {
		const a = {};
		return [a, ,];
	}
	const [x, ,] = f();
	t = x;
}
